cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_edge_case.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use crate::imports::*;
+use bridge_hub_westend_runtime::xcm_config::LocationToAccountId;
+use frame_support::traits::fungibles::Mutate;
+use snowbridge_router_primitives::inbound::EthereumLocationsConverterFor;
+use testnet_parachains_constants::westend::snowbridge::EthereumNetwork;
+use xcm_executor::traits::ConvertLocation;
+
+use crate::tests::snowbridge::{CHAIN_ID, ETHEREUM_DESTINATION_ADDRESS, WETH};
+
+const INITIAL_FUND: u128 = 5_000_000_000_000;
+const TOKEN_AMOUNT: u128 = 100_000_000_000;
+
+pub fn bridge_hub() -> Location {
+	Location::new(1, Parachain(BridgeHubWestend::para_id().into()))
+}
+
+pub fn beneficiary() -> Location {
+	Location::new(0, [AccountKey20 { network: None, key: ETHEREUM_DESTINATION_ADDRESS.into() }])
+}
+
+pub fn ethereum() -> Location {
+	Location::new(2, [GlobalConsensus(EthereumNetwork::get())])
+}
+
+pub fn weth_location() -> Location {
+	Location::new(
+		2,
+		[GlobalConsensus(EthereumNetwork::get()), AccountKey20 { network: None, key: WETH }],
+	)
+}
+
+pub fn fund_on_bh() {
+	let asset_hub_sovereign = BridgeHubWestend::sovereign_account_id_of(Location::new(
+		1,
+		[Parachain(AssetHubWestend::para_id().into())],
+	));
+	BridgeHubWestend::fund_accounts(vec![(asset_hub_sovereign.clone(), INITIAL_FUND)]);
+}
+
+pub fn fund_on_ah() {
+	AssetHubWestend::fund_accounts(vec![(AssetHubWestendSender::get(), INITIAL_FUND)]);
+	AssetHubWestend::fund_accounts(vec![(AssetHubWestendReceiver::get(), INITIAL_FUND)]);
+
+	AssetHubWestend::execute_with(|| {
+		assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::ForeignAssets::mint_into(
+			weth_location().try_into().unwrap(),
+			&AssetHubWestendReceiver::get(),
+			INITIAL_FUND,
+		));
+		assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::ForeignAssets::mint_into(
+			weth_location().try_into().unwrap(),
+			&AssetHubWestendSender::get(),
+			INITIAL_FUND,
+		));
+	});
+
+	let ethereum_sovereign: AccountId =
+		EthereumLocationsConverterFor::<[u8; 32]>::convert_location(&Location::new(
+			2,
+			[GlobalConsensus(EthereumNetwork::get())],
+		))
```

Comment:
You could use the `ethereum` helper function you made above.

```suggestion
		EthereumLocationsConverterFor::<[u8; 32]>::convert_location(&ethereum())
```

Reply:
Or maybe make this a helper function as well since you use it a few times

Reply:
https://github.com/paritytech/polkadot-sdk/pull/6838/commits/e792f7db45c507189165e924cbbd8ac4a226bf69